src/github.com/stellar/gateway/compliance/handlers/request_handler_tx_status.go

```diff
@@ -0,0 +1,91 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/stellar/gateway/protocols"
+	"github.com/stellar/gateway/server"
+	"github.com/stellar/go/protocols/compliance"
+)
+
+// HandlerTxStatus implements /tx_status endpoint
+func (rh *RequestHandler) HandlerTxStatus(w http.ResponseWriter, r *http.Request) {
```

Comment:
This method is way too nested to follow cleanly.  It's going to be a pain in the ass to maintain.